contracts/src/BeefyClient.sol

```diff
@@ -391,48 +420,37 @@ contract BeefyClient {
     }
 
     /**
-     * @dev Calculate minimum number of required signatures for the current validator set.
-     *
-     * This function approximates f(x) defined below for x in [1, 21_846):
-     *
-     *  x <= 10: f(x) = x * (2/3)
-     *  x  > 10: f(x) = max(10, ceil(log2(3 * x))
-     *
-     * Research by W3F suggests that `ceil(log2(3 * x))` is a minimum number of signatures required to make an
-     * attack unfeasible. We put a further minimum bound of 10 on this value for extra security.
+     * @dev Calculates the number of signature samples required for `submitFinal`.
+     * @param validatorSetLen The validator set length.
+     * @param signatureUseCount The count of how many times a validators signature was previously used in a call to `submitInitial`.
      *
-     * If the session has less than 10 active validators it's definitely some sort of local testnet
-     * and we use different logic (minimum 2/3 + 1 validators must sign).
+     *  ceil(log2(validatorSetLen)) + 1 * 2 ceil(log2(signatureUseCount))
      *
-     * One assumption is that Polkadot/Kusama will never have more than roughly 20,000 active validators in a session.
-     * As of writing this comment, Polkadot has 300 validators and Kusama has around 1000 validators,
-     * so we are well within those limits.
-     *
-     * In any case, an order of magnitude increase in validator set sizes will likely require a re-architecture
-     * of Polkadot that would make this contract obsolete well before the assumption becomes a problem.
-     *
-     * Constants generated with the help of scripts/minsigs.py
+     * See https://hackmd.io/9OedC7icR5m-in_moUZ_WQ for full analysis.
      */
-    function minimumSignatureThreshold(uint256 validatorSetLen) internal pure returns (uint256) {
-        if (validatorSetLen <= 10) {
+    function signatureSamples(uint256 validatorSetLen, uint256 signatureUseCount) internal view returns (uint256) {
```

Comment:
I think we should be clearer in our terminology. "Sampling" is the process of selecting the individuals in a subset. This function does not do that. Rather it calculates the size of the subset. So lets rename the function accordingly:

```suggestion
    function computeNumRequiredSignatures(uint256 validatorSetLen, uint256 signatureUsageCount) internal view returns (uint256) {
```

Reply:
Addressed in https://github.com/Snowfork/snowbridge/pull/956/commits/46250d8ff06a500a194a2c57acd59397e343af46